feat(cli): add codegen timeout scenario for pull failure (#5688)
aws-amplify/docs: src/pages/cli/project/troubleshooting.mdx

1.	[Backfill OpenSearch index from DynamoDB table](/cli/graphql/troubleshooting/#deploying-multiple-index-changes-at-once)
1.	[Index with Multi-Sort Key Fields](/cli/graphql/troubleshooting/#index-with-multiple-sort-key-fields)

## Troubleshooting "amplify pull" failures

### Pull failures with large number of UI components

> <Callout information>

Reviewer: does this render properly?

Reviewer: @vishal-dms and I touched base offline. It looks good!

Author: Yes

![amplify-pull-failure](https://github.com/aws-amplify/docs/assets/86995692/138a20d6-30d2-419d-846c-973195e496b5)

**Errors:**

"⠋ Generating UI components...🛑 Codegen job never succeeded before timeout"

> "✖ Failed to sync UI components"

Reviewer: can you put this in two separate fenced code blocks? This will help present the errors how they appear in the terminal with a monospace font.

Author: This format was to keep the document consistent as previous errors on the same page had similar styles

</Callout>

> <!-- cSpell:disable -->

Reviewer: what is the reason for disabling cspell?

Author: I got a spellcheck error for `CODEGENJOB` as follows -
```
> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
yarn run v1.22.19
$ git diff --name-only --cached | awk "/src.*\.mdx/{print}" | npx cspell --no-must-find-files --file-list stdin

1/1 ./src/pages/cli/project/troubleshooting.mdx 747.59ms X
/Documents/amplify-playground/docs/src/pages/cli/project/troubleshooting.mdx:225:222 - Unknown word (CODEGENJOB)
/Documents/amplify-playground/docs/src/pages/cli/project/troubleshooting.mdx:229:19 - Unknown word (CODEGENJOB)
CSpell: Files checked: 1, Issues found: 2 in 1 files
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
husky - pre-commit hook exited with code 1 (error)
```

Reviewer: Ah can you add that word to the dictionary to pass the spellcheck? https://github.com/aws-amplify/docs/blob/main/cspell.json#L4

Author: added `CODEGENJOB` to cspell.json

> Starting v12.2.0, Amplify CLI has introduced configurable timeout for generating UI components during `amplify pull`. By default this timeout is configured as 2 minutes but you can increase the same using the `UI_BUILDER_CODEGENJOB_TIMEOUT` environment variable as follows -

Reviewer: ```suggestion
Starting with v12.2.0, Amplify CLI has introduced a configurable timeout for generating UI components during `amplify pull`. By default this timeout is set to 2 minutes, however you can increase the timeout using the `UI_BUILDER_CODEGENJOB_TIMEOUT` environment variable.
```

Author: rephrased it in follow up commit
